client/src/app/auth/login.tsx

```diff
@@ -1,7 +1,52 @@
+import { ChangeEvent, FormEvent, useState } from "react";
+import Button from "../components/button";
+import TextField from "../components/text-field";
+
 export default function Login() {
   return (
     <div>
       <h1>Log in</h1>
+      <ExecLogIn />
     </div>
   );
 }
+
+
+function ExecLogIn() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleInputChange(e: ChangeEvent<HTMLInputElement>): void {
+    const { name, value } = e.target;
+    if (name === 'username') {
+      setUsername(value);
+    } else if (name === 'password') {
+      setPassword(value);
+    }
+  }
```

Comment:
We can combine the state of username and password into one object (and one interface using TypeScript)

Ex. 
```
interface LoginFormState {
  username: string;
  password: string;
}

const [formState, setFormState] = useState<LoginFormState>({ username: '', password: '' });

```

We then don't have to have this condition checking which field we are updating we can just use JavaScripts spread operator to update the whole object.